packages/consumption/src/modules/requests/utility/validateRelationshipAttributesWithinRequest.ts

```diff
@@ -0,0 +1,162 @@
+            `The Request can never be accepted because it would lead to the creation of more than one RelationshipAttribute in the context of this Relationship with the same key '${containsMustBeAcceptedDuplicatesResult.duplicateFragment.key}', owner and value type.`
+        );
+    }
+
+    const fragmentsOfAcceptedItemsOfRequest = extractRelationshipAttributeFragmentsFromAcceptedItems(items, params);
+
+    const containsAcceptedDuplicatesResult = containsDuplicateRelationshipAttributeFragments(fragmentsOfAcceptedItemsOfRequest);
+    if (containsAcceptedDuplicatesResult.containsDuplicates) {
+        return ValidationResult.error(
+            ConsumptionCoreErrors.requests.invalidAcceptParameters(
+                `The Request cannot be accepted with these parameters because it would lead to the creation of more than one RelationshipAttribute in the context of this Relationship with the same key '${containsAcceptedDuplicatesResult.duplicateFragment.key}', owner and value type.`
+            )
+        );
+    }
+
+    return ValidationResult.success();
+}
+
+function extractRelationshipAttributeFragmentsFromMustBeAcceptedItems(
+    items: (RequestItem | RequestItemGroup)[],
+    recipient?: CoreAddress
+): RelationshipAttributeFragment[] | undefined {
+    const fragmentsOfMustBeAcceptedItemsOfGroup: RelationshipAttributeFragment[] = [];
+
+    for (const item of items) {
+        if (item instanceof RequestItemGroup) {
+            const fragments = extractRelationshipAttributeFragmentsFromMustBeAcceptedItems(item.items, recipient);
+            if (fragments) fragmentsOfMustBeAcceptedItemsOfGroup.push(...fragments);
+        } else {
+            if (!item.mustBeAccepted) return;
+
+            const fragment = extractRelationshipAttributeFragmentFromRequestItem(item, recipient);
+            if (fragment) fragmentsOfMustBeAcceptedItemsOfGroup.push(fragment);
+        }
+    }
+
+    return fragmentsOfMustBeAcceptedItemsOfGroup;
+}
+
+function extractRelationshipAttributeFragmentsFromAcceptedItems(
+    items: (RequestItem | RequestItemGroup)[],
+    params: (DecideRequestItemParametersJSON | DecideRequestItemGroupParametersJSON)[]
+): RelationshipAttributeFragment[] | undefined {
+    const fragmentsOfAcceptedItemsOfRequest: RelationshipAttributeFragment[] = [];
+
+    for (let i = 0; i < items.length; i++) {
+        const item = items[i];
+        const decideItemParams = params[i];
+
+        if (item instanceof RequestItemGroup) {
+            const fragmentsOfAcceptedItemsOfGroup = extractRelationshipAttributeFragmentsFromAcceptedItems(
+                item.items,
+                (decideItemParams as DecideRequestItemGroupParametersJSON).items
+            );
+            if (fragmentsOfAcceptedItemsOfGroup) {
+                fragmentsOfAcceptedItemsOfRequest.push(...fragmentsOfAcceptedItemsOfGroup);
+            }
+        } else {
+            if (!(decideItemParams as DecideRequestItemParametersJSON).accept) return;
```

Comment:
If the condition is true, a return would be made and the `for` loop would be ended so that the remaining `items` are no longer processed. The condition should therefore be added to the one below.